src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.Windows.cs

```diff
@@ -42,68 +41,62 @@ public static Architecture OSArchitecture
         {
             get
             {
-                lock (s_osLock)
+                Debug.Assert(sizeof(Architecture) == sizeof(int));
+
+                if (s_osArch == -1)
```

Comment:
With s_osArch being volatile, it'd be good to avoid multiple reads of it on the fast path by storing it into a local.  While on x86/x64 volatile adds next to nothing, that's not the case on ARM.

Reply:
Another good point, #done